Use next_node blocks in calculate-your-child-maintenance

chrisroos opened the pull request:
We've agreed to consistently use `next_node {}` to define our next node rules. Having a single way of defining the rules will hopefully make Smart Answers easier to develop and maintain.

This will ultimately allow us to remove the predicate code (`define_predicate`, `on_condition`, `next_node_if` etc).

floehopper commented:
I think it would at least be good to explain your choice of `case` or `if/elsif` constructions in the PR description or commit messages.

floehopper commented:
Other than my very minor comments, this LGTM :+1:

chrisroos commented:
I've changed the `next_node` blocks for `:gross_income_of_payee?` and `:how_many_nights_children_stay_with_payee?` so that they use a `case` statement instead of `if .. else .. end`. This is more consistent with what I've done elsewhere, as highlighted by @floehopper.

I've rebased on master and force pushed in preparation for merging.
